Isolated removal

jbradberry opened the pull request:
##### SUMMARY
Removal of the isolated nodes feature.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```

```

jbradberry committed:
Chop out the dev environment isolated node

jbradberry committed:
Clean up the management commands

jbradberry committed:
Remove the IsolatedManager and its associated playbooks and plugins

AlanCoding commented on awx/main/management/commands/register_queue.py:
```diff
@@ -95,11 +80,6 @@ def register(self):
                     print("Instance Group already registered {}".format(ig.name))
 
-                if self.controller:
-                    (ig_ctrl, changed2) = self.update_instance_group_controller(ig)
```
You should get rid of the other uses of `changed2` here. It should work as it is, but we can get rid of the now-useless variable.

jbradberry replied:
This is all very much a work in progress.  Reviews are premature.

jbradberry committed:
Make the necessary changes to the models

- remove InstanceGroup.controller
- remove Instance.last_isolated_check
- remove .is_isolated and .is_controller methods/properties
- remove .choose_online_controller_node() method
- remove .supports_isolation() and replace with .can_run_containerized
- simplify .can_run_containerized

jbradberry committed:
Remove the isolation-specific settings

- AWX_ISOLATED_PUBLIC_KEY
- AWX_ISOLATED_PRIVATE_KEY
- AWX_ISOLATED_KEY_GENERATION
- AWX_ISOLATED_HOST_KEY_CHECKING
- AWX_ISOLATED_USERNAME
- AWX_ISOLATED_CONNECTION_TIMEOUT
- AWX_ISOLATED_LAUNCH_TIMEOUT
- AWX_ISOLATED_PERIODIC_CHECK
- AWX_ISOLATED_CHECK_INTERVAL

jbradberry committed:
Remove isolated nodes from the API views and serializers

AlanCoding commented on awx/main/models/jobs.py:
```diff
@@ -759,7 +755,7 @@ def display_artifacts(self):
     @property
     def can_run_containerized(self):
-        return any([ig for ig in self.preferred_instance_groups if ig.is_container_group])
+        return True
```
Calling out a notable change here. I agree with it, but will make sure to follow the use of the property to double-check.

AlanCoding replied:
Finished looking, I feel pretty confident in this.

jbradberry replied:
Grep shows 5 instances:

```
./awx/main/scheduler/task_manager.py:282:                elif task.can_run_cont>
./awx/main/scheduler/task_manager.py:485:                if task.can_run_contai>
./awx/main/models/jobs.py:757:    def can_run_containerized(self):
./awx/main/models/ad_hoc_commands.py:154:    def can_run_containerized(self):
./awx/main/models/unified_jobs.py:741:    def can_run_containerized(self):
```

3 are the defs, 1 is the one I know I moved this `any()` check to.  The other one (`./awx/main/scheduler/task_manager.py:485`) seems to be doing this same `.is_container_group` logic in a more complex loop.

jbradberry committed:
Update the docs to remove references to isolated nodes
